### DISCUSSION TITLE
how to get current location hash

### DISCUSSION THREAD

**Discussion by azim-b** [h2oai/wave#1275] [Q&A] [answered]
is there a way to get the current location hash from the app's url, within the app (ex: using meta)?

**Comment by mturoci** [comment, marked as answer]
`q.args['#']` should contain the current hash

**Reply by azim-b** [reply]
I thought this wouldn't work if the url is `localhost:10101/#demo`; i.e. `q.args['#']` would only contain current hash if a component with name containing hash is submitted with the form.

ex: `ui.button(name='#demo')`

Haven't tried this. Is it expected to work in both cases?

**Reply by mturoci** [reply]
Try it and you will see. Should be good for both :)
